base/combinatorics.jl

```diff
@@ -276,10 +276,10 @@ julia> B[invperm(v)]
 """
 function invperm(a::AbstractVector)
     require_one_based_indexing(a)
-    b = zero(a) # similar vector of zeros
+    b = fill!(similar(a), zero(eltype(a))) # similar vector of zeros
     n = length(a)
     @inbounds for (i, j) in enumerate(a)
-        ((1 <= j <= n) && b[j] == 0) ||
+        (1 <= j <= n) ||
```

Comment:
Why do you remove `b[j] == 0` check? I'm surprised this is not covered by CI.

Reply:
Oh, sorry about that. I misunderstood [this comment](https://github.com/JuliaLang/julia/pull/47855#discussion_r1044793858) not following the discussion.